app/controllers/ExternalServiceRedirectController.scala

```diff
@@ -36,10 +37,9 @@ class ExternalServiceRedirectController @Inject()(
   def auditInvalidateCacheAndRedirectService(serviceAndIFormName: String): Action[AnyContent] =
     (authenticate andThen validatePerson).async { implicit request =>
       {
-
         (for {
-          redirectUri <- auditService.sendAuditEventAndGetRedirectUri(request.taiUser.nino, serviceAndIFormName)
           _           <- sessionService.invalidateCache()
+          redirectUri <- auditService.sendAuditEventAndGetRedirectUri(request.taiUser.nino, serviceAndIFormName)
```

Comment:
might be worth modifying `sendAuditEventAndGetRedirectUri` instead of returning Future.successful it uses a `map`